### DISCUSSION TITLE
Where is a minimal example of Material.Avalonia ?

### DISCUSSION THREAD

**Discussion by buzzware** [AvaloniaCommunity/Material.Avalonia#360] [Q&A] [answered]
I'm struggling to understand the basics of this project.
It would help greatly to have a very simple example project that just includes the package, shows a few buttons and inputs with classes for different styles.
The main demo is too complicated - its hard to know what is its own code vs how you would use this package.

**Comment by SKProCH** [comment, marked as answer]
Hello, @buzzware 

Actually, this is a quite simple. 
1. Follow [Getting started](https://github.com/AvaloniaCommunity/Material.Avalonia#-getting-started) to setup library
2. Done. You can use controls as usual, and it will be styled properly.

You can actually use demo app to look how different styles look at demo app. Every control has the ShowMeTheXaml button which will show you a markup needed to recreate it.

![image](https://github.com/AvaloniaCommunity/Material.Avalonia/assets/29896317/24c9759e-ae04-46a3-ad2d-e9c150b0509b)

**Reply by SKProCH** [reply]
Also, the Getting Started wiki page has a bit more detailed guide: https://github.com/AvaloniaCommunity/Material.Avalonia/wiki/Getting-started

**Comment by buzzware** [comment]
Thanks, following https://github.com/AvaloniaCommunity/Material.Avalonia/wiki/Getting-started with a freshly generated project worked, and my project is here https://github.com/buzzware/AvaloniaMaterialTest
I've been developing an app with Avalonia without Material for about 6 months, and had years of Xamarin Forms experience before that. I'm on a Mac with Rider.
I have a very simple playground app for trying things out, and I could not get this project to work with it. I also couldn't get the demo project to build, or find a good intro to how to use the styles/themes. I wasn't sure whether to use Light or light, whether extra theme/style packages were required, how to access additional controls etc.
The demo code button is good, but how do you copy the code that appears? Do you type it off the screen?

**Reply by SKProCH** [reply]
> I have a very simple playground app for trying things out, and I could not get this project to work with it. I also couldn't get the demo project to build, or find a good intro to how to use the styles/themes.

Can you please provide a build errors? Probably something environment specific, cuz demo builds successfully on CI.

> I wasn't sure whether to use Light or light, whether extra theme/style packages were required, how to access additional controls etc.

By the way, this is all Avalonia specific, Material actually doesn't provide a custom really _complicated_ controls - just Avalonia.

Additional packages - only Material.Avalonia.DataGrid required to install, if you use DataGrid. Styles will be included automatically (for Material.Avalonia.DataGrid) in you use MaterialTheme in App.axaml.cs.

As for other packages or styles - you should refer it's docs, cuz, as I said earlier, there is no Material.Avalonia specific.

> The demo code button is good, but how do you copy the code that appears? Do you type it off the screen?

Just selecting text from the popup and copy paste it. Or type it off from the screen if you want, of course.

Probably most of the details should be in the wiki, but currently I doesn't have a time to fulfil the wiki (since fixing bugs and addressing issues is has more priority), sorry.

**Reply by SKProCH** [reply]
Also, as for your example, it runs fine for me

![image](https://github.com/AvaloniaCommunity/Material.Avalonia/assets/29896317/7311a48a-ed88-42bb-961d-eb2374d35093)

**Reply by buzzware** [reply]
As I said above, this new project works https://github.com/buzzware/AvaloniaMaterialTest
My playground project works too now - not sure what the problem was.
The demo project is building and running on desktop (Mac) and browser (Mac Chrome) now. The problem was a build error - some mac specific file missing. I did various things including building the Build project, reinstalling wasm and macos workload etc.

The demo project (https://avaloniacommunity.github.io/Material.Avalonia/) code display popup text on Mac Chrome has no cursor and is not selectable. It freezes the browser (Mac Chrome) and you get a "Page Unresponsive" popup.
On Desktop there is a cursor and I can copy and paste from it, but the normal selection text highlight is invisible, so it doesn't look like you can select it. The Reset button crashes the app, and the Apply button shows "Dynamic code generation is not supported on this platform." (which is properly highlighted on selection and does copy and paste).
On Mac Safari the code button freezes the browser without showing the popup (I have seen the popup appear only once).
I'll log an issue

**Reply by buzzware** [reply]
https://github.com/AvaloniaCommunity/Material.Avalonia/issues/361
